luarules/gadgets/game_team_power_watcher.lua

```diff
@@ -0,0 +1,514 @@
+                highestPower = power
+                highestTeamID = teamID
+            end
+        end
+    end
+
+    return {teamID = highestTeamID, power = highestPower}
+end
+
+--Returns the average of all non AI/scavenger/raptor teams as a number.
+local function averageHumanTeamPower()
+    local totalPower = 0
+    local teamCount = 0
+
+    for id, power in pairs(teamPowers) do
+        if humanTeams[id] then
+            totalPower = totalPower + power
+            teamCount = teamCount + 1
+        end
+    end
+
+    local averagePower = totalPower / teamCount
+    return averagePower
+end
+
+--Returns the lowest non AI/scavenger/raptor team power as a table {teamID, power}.
+local function lowestHumanTeamPower()
+    local lowestPower = math.huge
+    local lowestTeamID = nil
+
+    for teamID, power in pairs(teamPowers) do
+        if humanTeams[teamID] then
+            if power < lowestPower then
+                lowestPower = power
+                lowestTeamID = teamID
+            end
+        end
+    end
+
+    return {teamID = lowestTeamID, power = lowestPower}
+end
+
+--Returns the highest team power of the allies belonging to input team. Returns as a table {teamID, power}.
+local function highestAlliedTeamPower(teamID)
+    local allyTeamNum = select(6, Spring.GetTeamInfo(teamID))
+    local highestPower = 0
+    local highestTeamID = nil
+
+    for id, power in pairs(teamPowers) do
+        if allyTeamNum == select(6, Spring.GetTeamInfo(id)) then
+            if power > highestPower then
+                highestPower = power
+                highestTeamID = id
+            end
+        end
+    end
+
+    return {teamID = highestTeamID, power = highestPower}
+end
+
+--Returns the average of all allies of the input teamID. Returns a number.
+local function averageAlliedTeamPower(teamID) 
+    local allyTeamNum = select(6, Spring.GetTeamInfo(teamID))
+    local totalPower = 0
+    local teamCount = 0
+
+    for id, power in pairs(teamPowers) do
+        if allyTeamNum == select(6, Spring.GetTeamInfo(id)) then
+            totalPower = totalPower + power
+            teamCount = teamCount + 1
+        end
+    end
+
+    local averagePower = totalPower / teamCount
+    return averagePower
+end
+
+--Returns the lowest of the teamID's allies power as a table {teamID, power}.
+local function lowestAlliedTeamPower(teamID)
+    local allyTeamNum = select(6, Spring.GetTeamInfo(teamID))
+    local lowestPower = math.huge
+    local lowestTeamID = nil
+
+    for id, power in pairs(teamPowers) do
+        if allyTeamNum == select(6, Spring.GetTeamInfo(id)) then
+            if power < lowestPower then
+                lowestPower = power
+                lowestTeamID = id
+            end
+        end
+    end
+
+    return {teamID = lowestTeamID, power = lowestPower}
+end
+
+--Takes an input of a power value and compares it against powerThresholds table and returns an estimated tech level from 0.5-4.5.
```

Comment:
```suggestion
-- Take an input of a power value and return an estimated tech level.
```
Remove technical implementation details from function description, callers shouldn't need to be aware of them. Comment applies to other functions too.